api/proto/teleport/accesslist/v1/accesslist_service.proto

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.accesslist.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/accesslist/v1/accesslist.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1;accesslistv1";
+
+// AccessLisetService provides CRUD methods for Access List resources.
+service AccessListService {
+  // GetAccessLists returns a list of all access lists.
+  rpc GetAccessLists(GetAccessListsRequest) returns (GetAccessListsResponse);
+  // GetAccessList returns the specified access list resource.
+  rpc GetAccessList(GetAccessListRequest) returns (GetAccessListResponse);
+  // UpsertAccessList creates or updates an access list resource.
+  rpc UpsertAccessList(UpsertAccessListRequest) returns (google.protobuf.Empty);
+  // DeleteAccessList removes the specified access list resource.
+  rpc DeleteAccessList(DeleteAccessListRequest) returns (google.protobuf.Empty);
+  // DeleteAllAccessLists removes all access lists.
+  rpc DeleteAllAccessLists(DeleteAllAccessListsRequest) returns (google.protobuf.Empty);
```

Comment:
`DeleteAll` is part of all of our service definitions and it's largely used by the cache for clearing out local memory -- it's not typically exposed to the user.

Reply:
> and it's largely used by the cache for clearing out local memory

Reply:
> and it's largely used by the cache for clearing out local memory

Does it need to be in the RPC definitions or could this be a cache-only construct?

Reply:
I don't think this can be cache only since this needs to be part of the general `services.AccessLists` interface. Though I know this function is scary, I'm hesitant to deviate from the existing pattern:

```
➜  proto git:(mike.wilson/access-list-grpc) ✗ cat authservice.proto| grep "rpc DeleteAll"
  rpc DeleteAllNodes(types.ResourcesInNamespaceRequest) returns (google.protobuf.Empty);
  rpc DeleteAllApplicationServers(DeleteAllApplicationServersRequest) returns (google.protobuf.Empty);
  rpc DeleteAllAppSessions(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllSnowflakeSessions(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllSAMLIdPSessions(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllWebSessions(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllWebTokens(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllKubernetesServers(DeleteAllKubernetesServersRequest) returns (google.protobuf.Empty);
  rpc DeleteAllDatabaseServers(DeleteAllDatabaseServersRequest) returns (google.protobuf.Empty);
  rpc DeleteAllDatabaseServices(DeleteAllDatabaseServicesRequest) returns (google.protobuf.Empty);
  rpc DeleteAllServerInfos(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllApps(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllDatabases(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllKubernetesClusters(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllWindowsDesktopServices(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllWindowsDesktops(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllInstallers(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllSAMLIdPServiceProviders(google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc DeleteAllUserGroups(google.protobuf.Empty) returns (google.protobuf.Empty);
```